File: AspectsTests/TestObjects/TestObject.m
//
//  TestObject.m
//  AspectsTests
//
//  Created by Yanni Wang on 4/10/19.
//  Copyright © 2019 Yanni. All rights reserved.
//

#import "TestObject.h"

@implementation TestObject

- (void)methodWithExecuted:(nullable BOOL *)executed
{
    if (executed) {
        *executed = YES;
    }
}

- (void)methodWithExecutedBlock:(nullable void(^)(void))block
{
    if (block) {
        block();
    }
}

- (id)methodWithOriginalReturnValue:(id)value;
{
    return value;
}

@end

File: iOSHookTests/TestObject.swift
//
//  TestObject.swift
//  iOSHookTests
//
//  Created by Yanni Wang on 24/4/20.
//  Copyright © 2020 Yanni. All rights reserved.
//

import UIKit

class TestObject: NSObject {
    @objc dynamic func noArgsNoReturnFunc() {
        
    }
    
    @objc dynamic func sumFunc(a: Int, b: Int) -> Int {
        return a + b
    }
}
